src/openttd.cpp

```diff
@@ -659,6 +658,7 @@ int openttd_main(std::span<char * const> arguments)
 		BaseSounds::FindSets();
 		BaseMusic::FindSets();
 		ShowHelp();
+		if (i != 'h') ret = 1;
 		return ret;
```

Comment:
```suggestion
		/* Return zero when printing help, and 1 for all other cases. */
		if (i == 'h') return 0;
		return 1;
```

Feel free to write down any other comment; but something to indicate what is happening, would be nice.

Fun fact: I think with my two suggested changes, the variable `ret` will be unused. So you can remove that too :)